### PR TITLE
4.0.0

### PR DESCRIPTION
**Pull request by kirill-konshin**

### CODE REVIEW

**Comment by kirill-konshin** [comment]
Major / breaking changes planed for JS SDK 4.0 (also described in `CHANGELOG.md`):

- Remove magic endpoint prefix. In previous versions, endpoints such as `/account/~` will be prefixed with `/restapi/v1.0` magically. In 4.0, we do not do it any more. Users are required to specify the full endpoint.
- Pubnub was moved to separate package
- Async storage #94 
- Client ID & Secret naming #68

File: react/src/AuthGate.js [hunk 1]
```diff
@@ -0,0 +1,209 @@
+"use strict";
+var __extends = (this && this.__extends) || (function () {
```
Comment (embbnux) [inline react/src/AuthGate.js]: Why is this file not ES6?

Comment (kirill-konshin) [reply react/src/AuthGate.js]: Looks like some crap entered the PR, there supposed to be only TS file. I will clean this up.

Btw, you want ES6 along with ES5, I presume.

Comment (kirill-konshin) [reply react/src/AuthGate.js]: Removed those 2 files, seems that they were produced by IDE.

**Pull request merged**